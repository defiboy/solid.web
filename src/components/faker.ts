import { GenericType } from './fakeTypes'

export const buildFakeGenericType = (): GenericType[] => {
  const genericSamples: GenericType[] = [
    {
      name: 'Connection 1',
      url: 'http://localhost:8545'
    },
    {
      name: 'Connection 2',
      url: 'http://localhost:9000'
    }
  ]

  return genericSamples
}
