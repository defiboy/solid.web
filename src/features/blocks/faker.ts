import { Block } from "./types"

export const buildFakeBlock = (): Block => {
    return {
        difficulty: 54831103175033,
        extraData: "0xd783010406844765746887676f312e342e32856c696e7578",
        gasLimit: 4712388,
        gasUsed: 3059685,
        hash: "0xec4cd4f70ab16718ada2795b0b24f2be6498f4fca3e0b6b79e045933814f56e7",
        logsBloom: "0x00000000000000000400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000002000000000000000000000000000000000000000000000000002000000000000000000000000000000000",
        miner: "0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5",
        nonce: "0x8b119d486f87adf5",
        number: 1757201,
        parentHash: "0xd4bd6a69b51d768950f683151e009410b74403f8dc2780bf03697e3a6e4919c7",
        receiptRoot: "0x23ed6181c74edbcd2b29aa317dc84d320788a2c98b818b5fc49c475ef1d5164a",
        sha3Uncles: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
        size: 10736,
        stateRoot: "0x857cd117a16db6661eee8256702aa6879ecceccdb05a13637bad147760762547",
        timestamp: 1466691328,
        totalDifficulty: 30162772625997950877,
        transactions: ["0xd5875cdc5dea095e713b2409d585bd713d7f67bf1f0a0143293c00dc3db187da", "0xb7f5a83267df37248b44a7204b974a89d5ee870f00e91b6b2858b0088f8ada68", "0xc0fdbd563a0e26966e2f9803729f22bcaf29f0e73df158ccb0868058ce103ef2", "0xb294881e48ed00fe449b9604a29bea78db19ff1dee845f0af26eb75546a0a2f0", "0x0dda4a0efb859327c4755c9610bfafd921647542a60cd4d7fab37a2a1ac74b01", "0xbec6c60e5092138f52d0437c67528459d6a727332022b3686f1bc0df171a4c92", "0x5a9941189e91c5b1a68f8efd7a63b4aa28cb0fc72365d6bd4b2fabedd70da984", "0x13b66d8a1ce8eec4492e070037670c993477371c895e43d6dc498bc5d36ef726", "0x3db1158457a8659d9b249b06ccac059358eb110f9f136f229f94afb76144efdb", "0x15aa6a7e18237f6011bf5660c7c7fcdad57b00e2bd1b83179ed75c3a4d0521d3", "0xec830f03d41367e31d27578f1d981f45377accf0526ed9a9bc691e83b64694c2", "0x1d975af035632721617103e05bc0a4f035d9885ba8d279915dd9dee182a6c794", "0xbe0fc22dd5c7d5330cd5987ee63080d272c5802c7fee508602daf0220f3617df", "0x8dc9ef3427b16c240e6d4e2f4e428db575d5b55a89d22a392f4a0efa43110036", "0xd64b8106cbf8cd61288b765504125f68f9a74666a7aa2187bf83ac2a97cb6efd", "0x000c56bb6b29a1230cbceb3dfc165e15c040be8e5fcf1cb0c62397525bf528c1", "0x096f1b0ed9b7741582ee065bc9c725df0dbd5081fce37f1758c78c96028faa92", "0xed8af8503bfa73b1f5f92135712ef09c71db6e8e5b1a83fc2a5cd6e0fec5639a", "0x1e881a3643f099d84f34c3d5d0aa155cf737ea5348902d2675dec054e803cd11", "0x0289d236f7f9d14b82bd174cf317533566314231cb64c79ef1ec5532638b737b", "0xfa05d618ff5946eec57cb4770daa10f1751fed90a274688eccff2cb82c7365b5", "0x65fd378aec19b1e5302a14fac44077f4df13d5859f3fd27c86931c8ca0e5d0bd"],
        transactionRoot: "0xfe3316b06b1fad15e6e1c0e06ad2043213f36518f122ad6dff0293fe57d26d5f",
        uncles: []
    }
}

export const buildFakeBlocks = (): Block[] => {
    return [
        buildFakeBlock(),
        {
            difficulty: 54831103175033,
            extraData: "0xd783010406844765746887676f312e342e32856c696e7578",
            gasLimit: 4712388,
            gasUsed: 646893,
            hash: "0xdac2431708c0698b66a3a4a3ad4ba44b268df402b9350a909175d6ce6ca9004d",
            logsBloom: "0x00000000000000000400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000002000000000000000000000000000000000000000000000000002000000000000000000000000000000000",
            miner: "0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5",
            nonce: "0x8b119d486f87adf5",
            number: 646893,
            parentHash: "0xdac2431708c0698b66a3a4a3ad4ba44b268df402b9350a909175d6ce6ca9004e",
            receiptRoot: "0x23ed6181c74edbcd2b29aa317dc84d320788a2c98b818b5fc49c475ef1d5164a",
            sha3Uncles: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
            size: 10736,
            stateRoot: "0x857cd117a16db6661eee8256702aa6879ecceccdb05a13637bad147760762547",
            timestamp: 1466691328,
            totalDifficulty: 30162772625997950877,
            transactions: ["0xd5875cdc5dea095e713b2409d585bd713d7f67bf1f0a0143293c00dc3db187da", "0xb7f5a83267df37248b44a7204b974a89d5ee870f00e91b6b2858b0088f8ada68", "0xc0fdbd563a0e26966e2f9803729f22bcaf29f0e73df158ccb0868058ce103ef2", "0xb294881e48ed00fe449b9604a29bea78db19ff1dee845f0af26eb75546a0a2f0", "0x0dda4a0efb859327c4755c9610bfafd921647542a60cd4d7fab37a2a1ac74b01", "0xbec6c60e5092138f52d0437c67528459d6a727332022b3686f1bc0df171a4c92", "0x5a9941189e91c5b1a68f8efd7a63b4aa28cb0fc72365d6bd4b2fabedd70da984", "0x13b66d8a1ce8eec4492e070037670c993477371c895e43d6dc498bc5d36ef726", "0x3db1158457a8659d9b249b06ccac059358eb110f9f136f229f94afb76144efdb", "0x15aa6a7e18237f6011bf5660c7c7fcdad57b00e2bd1b83179ed75c3a4d0521d3", "0xec830f03d41367e31d27578f1d981f45377accf0526ed9a9bc691e83b64694c2", "0x1d975af035632721617103e05bc0a4f035d9885ba8d279915dd9dee182a6c794", "0xbe0fc22dd5c7d5330cd5987ee63080d272c5802c7fee508602daf0220f3617df", "0x8dc9ef3427b16c240e6d4e2f4e428db575d5b55a89d22a392f4a0efa43110036", "0xd64b8106cbf8cd61288b765504125f68f9a74666a7aa2187bf83ac2a97cb6efd", "0x000c56bb6b29a1230cbceb3dfc165e15c040be8e5fcf1cb0c62397525bf528c1", "0x096f1b0ed9b7741582ee065bc9c725df0dbd5081fce37f1758c78c96028faa92", "0xed8af8503bfa73b1f5f92135712ef09c71db6e8e5b1a83fc2a5cd6e0fec5639a", "0x1e881a3643f099d84f34c3d5d0aa155cf737ea5348902d2675dec054e803cd11", "0x0289d236f7f9d14b82bd174cf317533566314231cb64c79ef1ec5532638b737b", "0xfa05d618ff5946eec57cb4770daa10f1751fed90a274688eccff2cb82c7365b5", "0x65fd378aec19b1e5302a14fac44077f4df13d5859f3fd27c86931c8ca0e5d0bd"],
            transactionRoot: "0xfe3316b06b1fad15e6e1c0e06ad2043213f36518f122ad6dff0293fe57d26d5f",
            uncles: []
        }
    ]
}