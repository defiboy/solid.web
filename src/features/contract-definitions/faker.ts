import { ContractDefinition } from './types'

export const buildFakeContractDefinition = (): ContractDefinition => {
    const contractDefinition: ContractDefinition = {
        _id: '1',
        name: "ERC-20",
        sourceCode: "source code", // TODO
        abi: [], // TODO
        bytecode: "bytecode" // TODO
    }

    return contractDefinition
}

export const buildFakeContractDefinitions = (): ContractDefinition[] => {
    const connections: ContractDefinition[] = [{
        _id: '1',
        name: "ERC-20",
        sourceCode: "source code",
        abi: [],
        bytecode: ""
    },
    {
        _id: '2',
        name: "ERC-721",
        sourceCode: "source code",
        abi: [],
        bytecode: "bytecode"

    }]

    return connections
}
