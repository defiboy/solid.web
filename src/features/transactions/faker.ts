import { Transaction } from "./types"

export const buildFakeTransactions = (): Transaction[] => {
    return []
}