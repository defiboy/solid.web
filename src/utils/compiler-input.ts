const compilerInput = (sources: any, opts: any) => {
    return JSON.stringify({
        language: 'Solidity',
        sources: sources,
        settings: {
            optimizer: {
                enabled: opts.optimize === true || opts.optimize === 1,
                runs: 200
            },
            libraries: opts.libraries,
            outputSelection: {
                '*': {
                    '': ['legacyAST'],
                    '*': ['abi', 'metadata', 'devdoc', 'userdoc', 'evm.legacyAssembly', 'evm.bytecode', 'evm.deployedBytecode', 'evm.methodIdentifiers', 'evm.gasEstimates']
                }
            }
        }
    })
}
export default compilerInput