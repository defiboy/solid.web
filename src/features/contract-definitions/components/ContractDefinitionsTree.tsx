import React from 'react'
// import { Action, ActionCreator } from 'redux' Not needed maybe
import { Icon } from 'antd'

import { ContractDefinition } from '../types'
import { ContractDefinitionsTreeComponent } from './ContractDefinitionsTreeComponent'
import { TreeNodeStyled } from 'components/GenericTreeStyledComponents'

interface Props {
  contractDefinitions: ContractDefinition[]
}

const rightClickOptions = undefined
const onClickDataItem = undefined

export class ContractDefinitionsTree extends React.Component<Props> {
  render() {
    const { contractDefinitions } = this.props
    return (
      <ContractDefinitionsTreeComponent headerTitle="Contract Definitions"
        dataItems={contractDefinitions}
        onPlusClick={undefined}
        onCollapseClick={undefined}
        DataRowComponentRender={(item: ContractDefinition) => (
          <TreeNodeStyled
            icon={<Icon type="database" />}
            title={item.name}
            key={item.name}
            style={{ color: 'white' }}>
          </TreeNodeStyled>
        )}
        selectorPrefix="contract-definitions"
        onClickDataItem={onClickDataItem}
        rightClickMenuItems={rightClickOptions}
      />
    )
  }
}
