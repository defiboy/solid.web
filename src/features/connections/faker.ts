import { Connection } from './types'
import { generateFakeObjectId } from 'utils/fakeObjectId'
export const buildFakeConnection = (): Connection => {
    const connection: Connection = {
        name: "Connection 1",
        url: "http://localhost:8545",
        _id: generateFakeObjectId()
    }

    return connection
}

export const buildFakeConnections = (): Connection[] => {
    const connections: Connection[] = [
        buildFakeConnection(),
        {
            name: "Connection 2",
            url: "http://localhost:9000",
            _id: generateFakeObjectId()

        }]

    return connections
}
