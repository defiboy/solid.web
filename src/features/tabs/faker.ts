import { Tab } from './types'
import { generateFakeObjectId } from 'utils/fakeObjectId'

export const buildFakeTab = (): Tab => {
  const tab: Tab = {
    id: generateFakeObjectId(),
    type: 'transactions',
    data: [],
    title: 'Transactions'
  }

  return tab
}

export const buildFakeTabs = (): Tab[] => {
  const tabs: Tab[] = [
    buildFakeTab(),
    {
      id: generateFakeObjectId(),
      type: 'contracts',
      data: [],
      title: 'Contracts'
    }
  ]

  return tabs
}
