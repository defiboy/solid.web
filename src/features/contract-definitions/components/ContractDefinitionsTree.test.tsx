import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';

import { getMouseEvent } from 'utils/getMouseEvent';

import { buildFakeContractDefinitions } from '../faker'

import { ContractDefinitionsTree } from './ContractDefinitionsTree'

describe('ContractDefinitionsTree', () => {
    const onNewConnectionClickMockHandler = jest.fn()

    beforeEach(() => {
        onNewConnectionClickMockHandler.mockClear()
    })

    test('that it renders all the ui elements', () => {
        const contractDefinitions = buildFakeContractDefinitions()
        const text1 = contractDefinitions[0].name
        const text2 = contractDefinitions[1].name

        const { getByTestId, getByText } = render(
            <ContractDefinitionsTree contractDefinitions={contractDefinitions} />)

        expect(getByTestId('contract-definitions-tree-header')).toBeInTheDocument()
        expect(getByTestId('contract-definitions-tree-header')).toHaveTextContent("Contract Definitions")
        expect(getByTestId('contract-definitions-tree-plus')).toBeInTheDocument()
        expect(getByTestId('contract-definitions-tree-down')).toBeInTheDocument()

        expect(getByText(text1)).toBeInTheDocument()
        expect(getByText(text2)).toBeInTheDocument()
    })

    test('that no contract definitions are shown', () => {
        let contractDefinitions = buildFakeContractDefinitions()
        const text1 = contractDefinitions[0].name
        const text2 = contractDefinitions[1].name
        contractDefinitions = []

        const { getByTestId, queryByText } = render(
            <ContractDefinitionsTree contractDefinitions={contractDefinitions} />)

        expect(getByTestId('contract-definitions-tree-header')).toBeInTheDocument()
        expect(getByTestId('contract-definitions-tree-header')).toHaveTextContent("Contract Definitions")
        expect(getByTestId('contract-definitions-tree-plus')).toBeInTheDocument()
        expect(getByTestId('contract-definitions-tree-down')).toBeInTheDocument()

        expect(queryByText(text1)).not.toBeInTheDocument()
        expect(queryByText(text2)).not.toBeInTheDocument()
    })

    test('when right click is press in contract defintion, then the context menu should NOT appear', async () => {
        const contractDefinitions = buildFakeContractDefinitions()
        const text1 = contractDefinitions[0].name

        const { getByTestId, getByText } = render(
            <ContractDefinitionsTree contractDefinitions={contractDefinitions} />)

        const contractDefinitionFirstElement = getByText(text1)

        fireEvent.click(contractDefinitionFirstElement)

        const rightClick = getMouseEvent('contextmenu', {
            pageX: 250,
            pageY: 250,
            button: 2
        })

        fireEvent(contractDefinitionFirstElement, rightClick)

        await wait(() => {
            expect(getByTestId('contract-definitions-tree-rightclick-menu-option-openconsole')).not.toBeInTheDocument()
            expect(getByTestId('contract-definitions-tree-rightclick-menu-option-deploy')).not.toBeInTheDocument()
        })
    })

    test('renders contract definitions tree snapshop', async () => {
        const contractDefinitions = buildFakeContractDefinitions()

        const { container } = render(
            <ContractDefinitionsTree contractDefinitions={contractDefinitions} />)

        expect(container).toMatchSnapshot()
    })
})