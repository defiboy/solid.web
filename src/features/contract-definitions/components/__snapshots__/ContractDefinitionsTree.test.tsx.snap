// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractDefinitionsTree renders contract definitions tree snapshop 1`] = `
<div>
  <div
    style="overflow: scroll; height: 100%;"
  >
    <div />
    <div
      class="sc-bdVaJa cRJnvr"
    >
      <h3
        class="sc-bwzfXH boJNxd"
        data-testid="contract-definitions-tree-header"
      >
        Contract Definitions
      </h3>
      <div
        class="sc-htpNat KtFsv"
      >
        <i
          aria-label="icon: plus"
          class="anticon anticon-plus"
          data-testid="contract-definitions-tree-plus"
          style="color: white; padding-right: 0.5em;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </i>
        <i
          aria-label="icon: down"
          class="anticon anticon-down"
          data-testid="contract-definitions-tree-down"
          style="color: white;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </i>
      </div>
    </div>
    <ul
      class="ant-tree ant-tree-directory sc-bZQynM levuGt"
      role="tree"
      style="color: white;"
      unselectable="on"
    >
      <li
        class="sc-bxivhb iAyWlo ant-tree-treenode-switcher-open"
        role="treeitem"
        style="color: white;"
      >
        <span
          class="ant-tree-switcher ant-tree-switcher-noop"
        />
        <span
          class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
          title="ERC-20"
        >
          <span
            class="ant-tree-iconEle ant-tree-icon__customize"
          >
            <i
              aria-label="icon: file"
              class="anticon anticon-file"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="file"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0 0 42 42h216v494z"
                />
              </svg>
            </i>
          </span>
          <span
            class="ant-tree-title"
          >
            ERC-20
          </span>
        </span>
      </li>
      <li
        class="sc-bxivhb iAyWlo ant-tree-treenode-switcher-open"
        role="treeitem"
        style="color: white;"
      >
        <span
          class="ant-tree-switcher ant-tree-switcher-noop"
        />
        <span
          class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
          title="ERC-721"
        >
          <span
            class="ant-tree-iconEle ant-tree-icon__customize"
          >
            <i
              aria-label="icon: file"
              class="anticon anticon-file"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="file"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0 0 42 42h216v494z"
                />
              </svg>
            </i>
          </span>
          <span
            class="ant-tree-title"
          >
            ERC-721
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
